streaming/src/main/scala/org/apache/spark/streaming/scheduler/ReceiverTracker.scala

```diff
@@ -208,6 +208,7 @@ class ReceiverTracker(ssc: StreamingContext, skipReceiverLaunch: Boolean = false
   /** This thread class runs all the receivers on the cluster.  */
   class ReceiverLauncher {
     @transient val env = ssc.env
+    private var terminated = true
```

Comment:
Why is this initialized to true? Should it not be initialized to false. Then line 300 is not necessary. 
Also since this var is being modified and checked from multiple threads, best to mark this as volatile. And also transient.